### PR TITLE
Mat-4168 UMLS login dialog

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-4168](https://jira.cms.gov/browse/MAT-4168)
(Optional) Related Tickets:

### Summary
Initialize a "login session" with UMLS by hitting their api to retrieve an 8 hour TGT for future requests.
This TGT is stored in navigation for now.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Review by nmorasb — DISMISSED** [review]

File: src/components/MainNavBar/MainNavBar.tsx [hunk 1]
```diff
@@ -14,8 +14,29 @@ import { useOktaAuth } from "@okta/okta-react";
...
 import UserAvatar from "./UserAvatar";
+import UMLSDialog from "./ULMSDialog";
+import { Toast } from "@madie/madie-design-system/dist/react";
+
+import "./MainNavBar.scss";
 
 const MainNavBar = () => {
+  // dialog and toast utilities
+  const [TGT, setTGT] = useState<string>(""); //logged in url used to make requests
+  const [dOpen, setDOpen] = useState<boolean>(false);
```
Comment (adongare) [inline src/components/MainNavBar/MainNavBar.tsx:25]: `dOpen` is this a typo or intentional(some funky term like Toast)?

File: src/components/MainNavBar/MainNavBar.tsx [hunk 2]
```diff
@@ -58,11 +79,18 @@ const MainNavBar = () => {
...
                   </InnerItem>
                 </ListItem>
-
+                <ListItem className="activity-button">
+                  <button
+                    onClick={() => setDOpen(true)}
+                    data-testid="UMLS-connect-button"
+                  >
+                    <div className={TGT ? "active" : "inactive"} />
+                    {TGT ? "UMLS Active" : "Connect to UMLMS"}
```
Comment (adongare) [inline src/components/MainNavBar/MainNavBar.tsx]: looks like this is a typo too `UMLMS` vs `UMLS`

File: src/components/MainNavBar/ULMSDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  const onClose = () => {
+    handleClose();
+    formik.resetForm();
+  };
+  /* 
+    /cas/v1/api-key    Retrieves a Ticket Granting Ticket (TGT)
+    /cas/v1/tickets/{TGT}	Retrieves a single-use Service Ticket
+*/
+  const login = async (values: KeyConfig) => {
+    const url = "https://utslogin.nlm.nih.gov/cas/v1/api-key";
```
Comment (adongare) [inline src/components/MainNavBar/ULMSDialog.tsx:40]: I think this can be re-written using `https://vsac.nlm.nih.gov/vsac/ws` like @RohitKandimalla mentioned in VSAC spike presentation. it is easier. that way you don't have to parse the html.
url is `https://vsac.nlm.nih.gov/vsac/ws/Ticket  ` and `apikey={apikey}` is the request body. response body would be TGT

File: src/components/MainNavBar/MainNavBar.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/components/MainNavBar/MainNavBar.tsx:25]: Intentional short for dialog. It's the boolean for toast being put on the DOM

File: src/components/MainNavBar/ULMSDialog.tsx (hunk 3 above)
Comment (mcmcphillips) [reply src/components/MainNavBar/ULMSDialog.tsx:40]: This doesn't appear to work.
Regardless of allowing all cross origin headers, it doesn't pass the pre flight.
Do we have to manually allow that endpoint somewhere?

**Review by adongare — APPROVED** [review]
